lms/djangoapps/sga_api/views.py

```diff
@@ -0,0 +1,293 @@
+from student.views import create_account_with_params
+from xmodule.modulestore.django import modulestore
+
+log = logging.getLogger(__name__)
+
+
+class CreateUserAccountWithoutPasswordView(APIView):
+    """
+    Create user account without password
+    """
+    authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
+    permission_classes = (ApiKeyHeaderPermission,)
+
+    def post(self, request):
+        """
+        Create user account without password
+
+        Creates a user using mail, login and also name and surname.
+        Sets a random password and sends a user a message to change it.
+        """
+        data = request.data
+        data['honor_code'] = "True"
+        data['terms_of_service'] = "True"
+        email = request.data.get('email')
+        username = request.data.get('username')
+        # countries.by_name return country code or '' for else cases
+        country = countries.by_name(request.data.get('country', ''))
+        first_name = request.data.get('first_name', '')
+        last_name = request.data.get('last_name', '')
+        language = request.data.get('language')
+        if not username:
+            error_msg = "username is required parameter."
+        elif not email:
+            error_msg =  "email is required parameter."
+        elif not country:
+            error_msg = (
+                "Country is wrong or absent: {country}. For checking: Visit https://www.iso.org/obp . "
+                "Click the Country Codes radio option and click the search button."
+            ).format(country=request.data.get('country', ''))
+        elif language not in ['en', 'pt']:
+            error_msg = (
+                "language is wrong or absent: {language}. It must be:  pt (Portuguese) or en (English)"
+            ).format(language=language)
+        else:
+            error_msg = None
+
+        if error_msg:
+            return Response(
+                data={"error_message": error_msg},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        try:
+            validate_slug(username)
+        except ValidationError:
+            return Response(
+                data={
+                    "error_message": "Enter a valid 'username' consisting of letters, numbers, underscores or hyphens."
+                },
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        data['name'] = "{} {}".format(first_name, last_name).strip() if first_name or last_name else username
+
+        if check_account_exists(username=username, email=email):
+            return Response(data={"error_message": "User already exists"}, status=status.HTTP_409_CONFLICT)
+
+        try:
+            data['password'] = uuid4().hex
+            data['country'] = country
+            user = create_account_with_params(request, data)
+            user.is_active = True
+            user.save()
+            user_profile, _ = UserProfile.objects.get_or_create(user=user)
```

Comment:
create is redundant UserProfile is created on the create_account_with_params step and can be gotten by `user.profile`

Reply:
I changed them. Thank you for